hrm-domain/hrm-service/migrations/20231006194800-contact_taskId_unique.sql.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  up: async queryInterface => {
+    await queryInterface.sequelize.query(
+      `DELETE FROM "ContactJobs" WHERE "contactId" IN (SELECT id FROM "Contacts" WHERE "taskId" IN (SELECT "taskId" FROM public."Contacts" GROUP BY "taskId" HAVING COUNT(*) > 1));`,
+    );
+    await queryInterface.sequelize.query(
+      `DELETE FROM "CSAMReports" WHERE "contactId" IN (SELECT id FROM "Contacts" WHERE "taskId" IN (SELECT "taskId" FROM public."Contacts" GROUP BY "taskId" HAVING COUNT(*) > 1));`,
+    );
+    await queryInterface.sequelize.query(
+      `DELETE FROM "Referrals" WHERE "contactId" IN (SELECT id FROM "Contacts" WHERE "taskId" IN (SELECT "taskId" FROM public."Contacts" GROUP BY "taskId" HAVING COUNT(*) > 1));`,
+    );
+    await queryInterface.sequelize.query(
+      `DELETE FROM "ConversationMedias" WHERE "contactId" IN (SELECT id FROM "Contacts" WHERE "taskId" IN (SELECT "taskId" FROM public."Contacts" GROUP BY "taskId" HAVING COUNT(*) > 1));`,
+    );
+    await queryInterface.sequelize.query(
+      `DELETE FROM "Contacts" WHERE "taskId" IN (SELECT "taskId" FROM public."Contacts" GROUP BY "taskId" HAVING COUNT(*) > 1);`,
+    );
+    console.log('Cleared out contacts with duplicate taskIds');
```

Comment:
Question: wouldn't this delete both "the correct contact" and the "duplicated contacts"?
When we find duplicated contacts, I imagine we want to keep one and delete the others.

Reply:
We should manually check the data before running this migration, because we have no deterministic way to decide which contact we want to keep in a given script. 

So really there shouldn't be any dups in production data we run this migration on